cmd/submit_test.go

```diff
@@ -272,7 +272,7 @@ func TestSubmitFiles(t *testing.T) {
 	if assert.NoError(t, err) {
 		assert.Equal(t, 3, len(submittedFiles))
 		assert.Equal(t, "This is file 1.", submittedFiles["file-1.txt"])
-		assert.Equal(t, "This is file 2.", submittedFiles["subdir/file-2.txt"])
+		assert.Equal(t, "This is file 2.", submittedFiles["file-2.txt"])
```

Comment:
Hey! I'm a bit unclear on this test changing. If someone submits:
```
file-1.txt
subdir/file-2.txt 
file-2.txt
```

what happens? Have we lost that ability, or just lost a test for that? Or something else? :)

Reply:
This change is just to fix the response in the test case. As the multipart form used in the test server now only returns the filename without the directory information [link to Go source]( https://cs.opensource.google/go/go/+/refs/tags/go1.21.1:src/mime/multipart/multipart.go;l=99)

**Using Go 1.20.8**

This is the fileheader information that gets returned in the test. You can see the form-data is correct, which will be handled by the upstream API as is. But the returned Filename contains just the base filename, which is what is causing the test to fail. 
```shell
=== RUN   TestSubmitFiles
{
  "Filename": "file-1.txt",
  "Header": {
   "Content-Disposition": [
    "form-data; name=\"files[]\"; filename=\"file-1.txt\""
   ],
   "Content-Type": [
    "application/octet-stream"
   ]
  },
  "Size": 15
 },
{
  "Filename": "file-2.txt",
  "Header": {
   "Content-Disposition": [
    "form-data; name=\"files[]\"; filename=\"subdir/file-2.txt\""
   ],
   "Content-Type": [
    "application/octet-stream"
   ]
  },
  "Size": 15
 },
{
  "Filename": "README.md",
  "Header": {
   "Content-Disposition": [
    "form-data; name=\"files[]\"; filename=\"README.md\""
   ],
   "Content-Type": [
    "application/octet-stream"
   ]
  }
--- FAIL: TestSubmitFiles (0.00s)
```

With the fix in place you see the fileheader information is still the same but the test is now just expecting the base filename so all is well. 
```shell
=== RUN   TestSubmitFiles

{
  "Filename": "file-1.txt",
  "Header": {
   "Content-Disposition": [
    "form-data; name=\"files[]\"; filename=\"file-1.txt\""
   ],
   "Content-Type": [
    "application/octet-stream"
   ]
  },
  "Size": 15
 },
{
  "Filename": "file-2.txt",
  "Header": {
   "Content-Disposition": [
    "form-data; name=\"files[]\"; filename=\"subdir/file-2.txt\""
   ],
   "Content-Type": [
    "application/octet-stream"
   ]
  },
  "Size": 15
 },
{
  "Filename": "README.md",
  "Header": {
   "Content-Disposition": [
    "form-data; name=\"files[]\"; filename=\"README.md\""
   ],
   "Content-Type": [
    "application/octet-stream"
   ]
  },
  "Size": 19
 }
-- PASS: TestSubmitFiles (0.00s)
```

Reply:
> With the fix in place you see the fileheader information is still the same but the test is now just expecting the base filename so all is well.

I see, but we _are_ losing information now, as we're no longer testing the subdir, which is really important. Can the test be restructured to allow testing the filename _and_ the directory?

Reply:
Ah that's a fair point. I was thinking solely from the point of the submitted file content and not the directory tree. We can pull the unmodified filename from the Header to validate instead of using fileHeader.Filename. Which looking at the Go PR where this change was introduced implies as the thing to do. I opened a PR to with a fix that keeps the directory info intact. Refer to https://github.com/exercism/cli/pull/1115